File: REST-SOAP/src/main/java/dk/ek/gameapi/entity/Game.java
package dk.ek.gameapi.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "game")
@Getter
@Setter
@NoArgsConstructor
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(nullable = false)
    private String name;
    private String slug;
    @Column(name = "year_published", columnDefinition = "YEAR")
    private Integer yearPublished;
    private Double bggRating;
    private Double difficultyRating;
    @Column(columnDefinition = "TEXT")
    private String description;
    private Integer playingTime;
    private Boolean available;
    private Integer minPlayers;
    private Integer maxPlayers;
    private Integer minimumAge;
    @Column(columnDefinition = "TEXT")
    private String thumbnail;
    @Column(columnDefinition = "TEXT")
    private String image;
}

File: GraphQL/src/types/Game.ts
import { ObjectType, Field, Int, Float } from "type-graphql";

@ObjectType()
export class Game {
  @Field(() => Int)
  id: number;

  @Field()
  name: string;

  @Field({ nullable: true })
  slug?: string;

  @Field(() => Int, { nullable: true })
  year_published?: number;

  @Field(() => Float, { nullable: true })
  bgg_rating?: number;

  @Field(() => Float, { nullable: true })
  difficulty_rating?: number;

  @Field({ nullable: true })
  description?: string;

  @Field(() => Int, { nullable: true })
  playing_time?: number;

  @Field({ nullable: true })
  available?: boolean;

  @Field(() => Int, { nullable: true })
  min_players?: number;

  @Field(() => Int, { nullable: true })
  max_players?: number;

  @Field(() => Int, { nullable: true })
  minimum_age?: number;
}